lib/web/ui/usercontext.go

```diff
@@ -1,43 +1,91 @@
 package ui
 
 import (
+	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/services"
-
-	"github.com/gravitational/trace"
+	"github.com/gravitational/teleport/lib/utils"
 )
 
+type access struct {
+	Read   bool `json:"read"`
+	Edit   bool `json:"edit"`
+	Create bool `json:"create"`
+	Delete bool `json:"remove"`
+}
+
+type userACL struct {
+	// Sessions defines access to recorded sessions
+	Sessions access `json:"sessions"`
+	// AuthConnectors defines access to auth.connectors
+	AuthConnectors access `json:"authConnectors"`
+	// Roles defined access to roles
+	Roles access `json:"roles"`
+	// TrustedClusters defined access to trusted clusters
+	TrustedClusters access `json:"trustedClusters"`
+	// SSH defined access to servers
+	SSHLogins []string `json:"sshLogins"`
+}
+
 type userContext struct {
 	// Name is this user name
 	Name string `json:"userName"`
 	// Email is this user email
 	Email string `json:"userEmail"`
-	// ACL is this user access control list
-	ACL RoleAccess `json:"userAcl"`
+	// Logins is this user available logins
+	Logins []string `json:"userLogins"`
+	// ACL contains user access control list
+	ACL userACL `json:"userAcl"`
 }
 
-// NewUserContext returns userContext
-func NewUserContext(user services.User, allRoles []services.Role) (*userContext, error) {
-	userRoles := user.GetRoles()
-	roleNamesMap := map[string]bool{}
-	for _, name := range userRoles {
-		roleNamesMap[name] = true
+func getLogins(roleSet services.RoleSet) []string {
+	allLogins := []string{}
+	for _, role := range roleSet {
+		logins := role.GetLogins(services.Allow)
```

Comment:
Just a heads up, you are only requesting the allow logins, a login in an allow in one role could be a deny in another.